guibot/path.py

```diff
@@ -113,3 +113,47 @@ def search(self, filename, restriction="", silent=False):
             raise FileNotFoundError('File ' + filename + ' not found')
 
         return None
+
+    def __iter__(self):
+        for p in self._target_paths:
+            yield p
+
+    def __len__(self):
+        return len(self._target_paths)
```

Comment:
Useful additions, do we make use of them in a unit test?

Reply:
Indirectly. This is used by the `ScopedPath` class to copy the existing paths.